### ISSUE TITLE
Hisat2 align on large index

### ISSUE THREAD

**Issue by andreott** [snakemake/snakemake-wrappers#2537]
**Snakemake version**
*snakemake-wrappers:* 3.3.3 

**Describe the bug**
The wrapper looks for the index file checking for *.ht2* file extension only. So this fails when a *large* index was produced with extension *.ht2l* 

**Fixing**
A quick and dirty fixing would just use 
```python
ht2_files = Path(snakemake.input.idx).glob("*.ht2*")
```
 instead of 
```python
ht2_files = Path(snakemake.input.idx).glob("*.ht2")
```

... or search for both and combine

However I would rather vote for the user to provide the prefix instead of searching in the provided directory, as problems would arise with multiple indexes in the provided path.

**Comment by fgvieira** [comment]
I agree. The user providing the prefix sounds safer. Do you think you can make a PR with those changes?

**Comment by andreott** [comment]
Yes, I will provide a PR

**Comment by fgvieira** [comment]
Actually, I think it might be better to just provide all the index files as input. Something like:
```
rule hisat2_align:
    input:
        reads=["reads/{sample}_R1.fastq", "reads/{sample}_R2.fastq"],
        idx=multiext("index/ref", ".1.ht2", ".2.ht2", ".3.ht2", ".4.ht2", ".5.ht2", ".6.ht2", ".7.ht2", ".8.ht2"),
[...]
```

Then, I think you just need to delete line:
https://github.com/snakemake/snakemake-wrappers/blob/caa3b3ffaab9aa95cf9d7818c882e789255b9e9b/bio/hisat2/align/wrapper.py#L29

And fix line:
https://github.com/snakemake/snakemake-wrappers/blob/caa3b3ffaab9aa95cf9d7818c882e789255b9e9b/bio/hisat2/align/wrapper.py#L30

**Comment by andreott** [comment]
I think the problem is, if you build the index within the workflow, you will not know in advance if it is a large index.

**Comment by fgvieira** [comment]
Unless you force `hisat2-build` to build a [large index](https://daehwankimlab.github.io/hisat2/manual/#hisat2-build-options-large-index).

**Comment by andreott** [comment]
Hi, getting back to this one - sorry was quite busy the last weeks. Well for me, both are ok. As for the STAR wrapper, I guess it would be more portable not to provide the exact filenames and forcing the user to build the large index to enforce portability. Also I realised that my previous concern (multiple indices in the same directory) is not valid, as the wrapper is supposed to be called with the directory as output, so it will be overwritten by the next indexing run.
